lib/line-protocol.js

```diff
@@ -6,181 +6,212 @@
  */
 
 // Load modules
-const Querystring = require('querystring');
+const _ = require('lodash');
+const Os = require('os');
 const Url = require('url');
-const Hoek = require('hoek');
-const OpsFormatter = require('./ops-format');
 const Helpers = require('./helpers');
 
 const Formatters = require('./formatters');
 
-const internals = {};
+const formatSchema = (event, schema, config) => {
+    const lines = [];
+    if (schema.splitLines) {
+        const base = _.get(event, schema.splitLines.splitOn);
+        Object.keys(base).forEach((key) => {
+            const dataObj = {};
+            if (schema.splitLines.splitKey) {
+                dataObj[schema.splitLines.splitKey] = key;
+            }
+
+            const chunk = formatSchemaChunk(event, schema, config, dataObj);
 
-internals.formatError = (error) => {
+            if (chunk) {
+                lines.push(chunk);
+            }
+        });
+    }
+    else {
+        const chunk = formatSchemaChunk(event, schema, config, {});
 
-    const result = {
-        'error.name': Formatters.String(error.name),
-        'error.message': Formatters.String(error.message),
-        'error.stack': Formatters.String(error.stack)
+        if (chunk) {
+            lines.push(chunk);
+        }
+    }
+
+    return lines;
+};
+
+const formatSchemaChunk = (event, schema, config, data) => {
+    const timestamp = event.timestamp;
+    let metric = schema.metric;
+    let tags = {
+        host: event.host || Os.hostname(),
+        pid: event.pid
     };
+    let fields = formatData(event, schema.fields, data, false);
 
-    if (error.isBoom) {
-        result['error.statusCode'] = Formatters.Int(error.output.statusCode);
-        result.error = Object.assign({}, Formatters.Flatten(error.data), result.error);
+    fields = _.omitBy(fields, _.isUndefined);
+
+    if (_.isEmpty(fields)) {
+        return;
     }
 
-    return result;
+    if (schema.tags) {
+        tags = Object.assign({},
+            tags,
+            formatData(event, schema.tags, data, true)
+        );
```

Comment:
Ooooh, I didn't realize that. Hah. Fixed!

Reply:
I'd imagine it's a slight optimization to copy values from one object to another instead of copying values from two both objects into another, which then gets assigned to the same variable 😄